### PR TITLE
Updated GSL orographic drag suite to enable use of custom orographic …

### PR DESCRIPTION
**Pull request by mdtoy**
This is an update of the GSL orographic drag suite to use separate "small-scale" and "large-scale" orographic drag parameters (varss, olss, etc.), both of which can be read in by custom created static files.

### CODE REVIEW

File: physics/drag_suite.F90 [hunk 1]
```diff
@@ -474,7 +474,16 @@ subroutine drag_suite_run(                                           &
    errmsg = ''
    errflg = 0
 
-if (me==master) print *,"Running drag suite"
+
```
Comment (climbfuji) [inline physics/drag_suite.F90:477]: Please remove lines 477-484.

File: physics/drag_suite.F90 [hunk 2]
```diff
@@ -307,9 +307,10 @@ subroutine drag_suite_run(                                           &
    real(kind=kind_phys) ::  rcl, cdmb
    real(kind=kind_phys) ::  g_inv
 
-   real(kind=kind_phys), intent(out) ::                          &
+   real(kind=kind_phys), intent(inout) ::                        &
```
Comment (climbfuji) [inline physics/drag_suite.F90:310]: Please check if this change is required, and if so, please check and change the corresponding intent information in `drag_suite.meta`, if necessary.

File: physics/GFS_GWD_generic.F90 [hunk 3]
```diff
@@ -29,7 +30,8 @@ subroutine GFS_GWD_generic_pre_run(                               &
       real(kind=kind_phys), intent(in) :: mntvar(im,nmtvr)
 
       real(kind=kind_phys), intent(out) ::                              &
-     &  oc(im), oa4(im,4), clx(im,4),                                   &
+     &  var(im), oc(im), oa4(im,4), clx(im,4),                          &
```
Comment (climbfuji) [inline physics/GFS_GWD_generic.F90]: Are all these variables allocated all the time, or some only if the drag suite if used? If the latter is the case, these variables must have assumed-size definitions, e.g. something like `foo(:,:)`.

**Review by climbfuji — CHANGES_REQUESTED** [review]
A couple of questions and small changes. Does this PR change the answer for other physics as well, or just for the drag suite?

File: physics/drag_suite.F90 (hunk 1 above)
Comment (mdtoy) [reply physics/drag_suite.F90:477]: Lines 477-484 have been removed.

File: physics/drag_suite.F90 (hunk 2 above)
Comment (mdtoy) [reply physics/drag_suite.F90:310]: Actually, in drag_suite.meta, the three variables dudt, dvdt and dtdt had always been designated "intent = inout".  This is the correct designation, and I was bringing the drag_suite.F90 intent information into agreement.

File: physics/GFS_GWD_generic.F90 (hunk 3 above)
Comment (mdtoy) [reply physics/GFS_GWD_generic.F90]: When the GSL drag suite is used, var(im), varss(im), ocss(im), oa4ss(im,4) and clxss(im,4) are additional variables.  The variables oc(im), oa4(im), oa4(im,4) and clx(im,4) are used by both the GSL drag suite and the EMC drag suite.

**Comment by mdtoy** [comment]
Dom,

In answer to your question:
"Does this PR change the answer for other physics as well, or just for the drag suite?"

This PR only changes the answer for the drag suite.

-- Mike

File: physics/GFS_GWD_generic.F90 (hunk 3 above)
Comment (climbfuji) [reply physics/GFS_GWD_generic.F90]: The variables that are only allocated when the drag suite is used (see logic in GFS_typedefs.F90) need to be declared as asssumed-size arrays.

**Pull request merged**